[Liu Zhengyang] iP

redcolorbicycle opened the pull request:
# Bearducky

>This is making me feel like dropping out    -Zhengyang

Bearducky is:
- Text-based
- Easy to read
- **A duck wearing a bear hat!**
- ~~The reason why I know I will probably not be an SWE~~ 🆘 

Bearducky can:
1. Manage Tasks
2. Manage Deadlines

Here's a look at the main method:

```java
public static void main(String[] args) {
        Duke bearducky = new Duke("./data/tasks.txt");
        bearducky.run();
}
```

Coding this has made me (take a guess):
- [x] Cry
- [x] Break my computer
- [x] Fully fill out a dropout form but not submit it

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

redcolorbicycle committed:
Level 0 Rename

redcolorbicycle committed:
Level 1 Echo

redcolorbicycle committed:
Level 2 Add, List

redcolorbicycle committed:
Task class shell

redcolorbicycle committed:
Level 3 Mark/Unmark

redcolorbicycle committed:
Level 3 Mark/Unmark

redcolorbicycle committed:
Level 4 Todo Event Deadline

redcolorbicycle committed:
A-TextUiTesting

redcolorbicycle committed:
Handle basic errors

redcolorbicycle committed:
Deleting

redcolorbicycle committed:
halfway through level 7

redcolorbicycle committed:
halfway through level 7

redcolorbicycle committed:
halfway through level 7

redcolorbicycle committed:
halfway through level 8

redcolorbicycle committed:
Merge branch 'branch-level-7'

redcolorbicycle committed:
Level 8

redcolorbicycle committed:
Merge branch 'branch-Level-8'

redcolorbicycle committed:
A-MoreOOP

redcolorbicycle committed:
A-MoreOOP

redcolorbicycle committed:
A-Packages

redcolorbicycle committed:
Merge remote-tracking branch 'origin/add-gradle-support'

redcolorbicycle committed:
A-Gradle

redcolorbicycle committed:
A-JUnit

redcolorbicycle committed:
A-JavaDoc

redcolorbicycle committed:
branch-A-JavaDoc

redcolorbicycle committed:
A-JavaDoc

redcolorbicycle committed:
A-CodingStandard

redcolorbicycle committed:
Level-9

redcolorbicycle committed:
Resolved conflicts

redcolorbicycle committed:
Resolved conflicts

Shauryan123 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,311 @@
+    }
+
+
+    public LocalDate DateConvert(String s) {
```
```suggestion
  public LocalDate convertDate(String s) {
    // Implementation remains the same
}

```

Shauryan123 replied:
Rename DateConvert to convertDate to follow camelCase naming conventions for methods.

Shauryan123 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,311 @@
+            return;
+
+        } catch (ArrayIndexOutOfBoundsException b) {
+            System.out.println("Please enter a event with the format event eventName /from dd/mm/yyyy /to dd/mm/yyyy!");
```
Abstract out the ui to show the error rather than hardcoding it in the method
```suggestion
         ui.showError("Event cannot be blank");

```

Shauryan123 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,311 @@
+     * @param s The user input command string for adding an event task.
+     * @param t The task list to which the event task will be added.
+     */
+    public void handleEvent(String s, TaskList t) {
```
Refactor validation logic into separate methods to improve readability and maintainability. For example, extracting the validation of event dates into a separate method would clarify the handleEvent method.

qinboan commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,56 @@
+     * Displays an instruction message for the user.
+     */
+    public void instructionMessage() {
+        System.out.println("[quack] I don't understand that command.\n If you would like to add a task to the Duckalendar," +
+                " please specify with the task type - \"todo\", \"deadline\" or \"event\" then space then task" +
+                " name.\n For deadlines, please add a /by followed by the deadline (eg. /by 2/12/2019 1800) . For events, please add a / followed by the start time,\n" +
+                " then another \"/\" followed by the end time.\n" +
+                " To mark or unmark the nth task on the list, enter the command \"mark n\" or \"unmark n\" where n is the number on the list.\n" +
+                " To delete the nth task on the list, enter the command \"delete n\" where n is the number on the list.\n" +
+                " If you would like me to list the things you are procrastinating, please enter the word \"list\"." +
+                " \n If you would like to leave, please enter the word \"bye\". \n Do also feel free to also type the words \"feed bread to bearducky\"?" +
+                " [Hopeful quacking]");
```
Might want to break this into different println to improve readability

qinboan commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,147 @@
+        boolean marked = inputs[1].equals("marked");
+        DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+        switch (type) {
+            case "T":
```
there shouldnt be any indentation for case clauses

qinboan commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,311 @@
+                    " only, please.");
+        }
+    }
+}
```
nice work handling with user commands with the various methods!

qinboan commented on src/test/java/duke/StorageTest.java:
```diff
@@ -0,0 +1,16 @@
+
+public class StorageTest {
+
+    @Test
```
maybe you could consider testing some other cases such as checking whether todo, deadline and event are correctly stored in the storage?

qinboan reviewed (commented):
LGTM

redcolorbicycle committed:
Level-10

redcolorbicycle committed:
Fixed error

redcolorbicycle committed:
Assertions added to some methods

A few methods have assumptions, such as that the string word input will
be one of the few options.

Adding assertions helps clarify the error

Assertion code has been added

redcolorbicycle committed:
Merge pull request #2 from redcolorbicycle/branch-A-Assertions

Assertions added to some methods

redcolorbicycle committed:
Added C-BetterSearch

Find only returned one task that closely matched input.

Function has been modified to return all tasks that contain the input in
chronological order, even if separated by alphabets.

redcolorbicycle committed:
Checkstyle

redcolorbicycle committed:
Update README.md

redcolorbicycle committed:
Update README.md

redcolorbicycle committed:
Update README.md

redcolorbicycle committed:
Changed format error: dd/mm should be mm/dd

redcolorbicycle committed:
Merge branch 'master' of https://github.com/redcolorbicycle/ip

redcolorbicycle committed:
Add files via upload

redcolorbicycle committed:
Update README.md

redcolorbicycle committed:
Update README.md

redcolorbicycle committed:
Final Release

redcolorbicycle committed:
Final Release
